dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/guestbook/ManageGuestbooksCRUDServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package edu.harvard.iq.dataverse.guestbook;
+
+import edu.harvard.iq.dataverse.DataverseRequestServiceBean;
+import edu.harvard.iq.dataverse.EjbDataverseEngine;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.impl.DeleteGuestbookCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseCommand;
+import edu.harvard.iq.dataverse.engine.command.impl.UpdateDataverseGuestbookRootCommand;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+import edu.harvard.iq.dataverse.persistence.guestbook.Guestbook;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@MockitoSettings(strictness = Strictness.LENIENT)
+@RunWith(MockitoJUnitRunner.class)
+public class ManageGuestbooksCRUDServiceTest {
+    @InjectMocks
+    private ManageGuestbooksCRUDService manageGuestbooksCRUDService;
+
+    @Mock
+    private EjbDataverseEngine engineService;
+    @Mock
+    private DataverseRequestServiceBean dvRequestService;
+
+
+    @BeforeEach
+    void setUp() throws CommandException {
+        when(engineService.submit(Mockito.any(UpdateDataverseCommand.class))).thenReturn(new Dataverse());
+        when(engineService.submit(Mockito.any(DeleteGuestbookCommand.class))).thenReturn(new Dataverse());
+        when(engineService.submit(Mockito.any(UpdateDataverseGuestbookRootCommand.class))).thenReturn(new Dataverse());
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void createOrUpdate() throws CommandException {
+        // given
+        Dataverse dv = createTestDataverse();
+
+        // when
+        manageGuestbooksCRUDService.createOrUpdate(dv);
+
+        // then
+        verify(engineService, times(1)).submit(Mockito.any(UpdateDataverseCommand.class));
+    }
+
+    @Test
+    public void createOrUpdate_withUpdateException() throws CommandException {
+        // given
+        Dataverse dv = createTestDataverse();
+
+        // when
+        when(engineService.submit(Mockito.any(UpdateDataverseCommand.class))).thenAnswer(Answers.CALLS_REAL_METHODS);
```

Comment:
`Answers.CALLS_REAL_METHODS` is kind of contradiction of unit tests purpouse (should be probably used in very specific situations).
If you need for something to throw exception then simply use `thenThrow`

Reply:
wouldn't then the test be kinda pointless if we mock tested function?